caluma/caluma_form/domain_logic.py

```diff
@@ -200,3 +200,24 @@ def create(
     @transaction.atomic
     def update(document, **kwargs):
         update_model(document, kwargs)
+
+
+class CopyFormLogic:
+    @staticmethod
+    def copy(validated_data: dict, user: Optional[BaseUser] = None):
+        source = validated_data["source"]
+        validated_data["meta"] = dict(source.meta)
+        form = BaseLogic.create(model=models.Form, user=user, **validated_data)
+
+        for sort, form_question in enumerate(
+            reversed(models.FormQuestion.objects.filter(form=source)), start=1
```

Comment:
Use `.order_by("sort")` instead of `reversed` to make it more clear what this is about.

Reply:
Actually, this whole sorting is useless :rofl: I now set `sort` to `form_question.sort` which is what we want here.

Reply:
That's also correct, but you might have gaps in the `sort`. With the previous method there would be nice refresh sequence 1..n.

Reply:
Not really, those are being "refreshed" every time one of them change. Also, it's supposed to be a 1:1 copy.